### DISCUSSION TITLE
How can I pick out a subset of grains with GOS less than 2° for IPF ?

### DISCUSSION THREAD

**Discussion by SingPr** [mtex-toolbox/mtex#1543] [Ask Anything] [answered]
Dear community,
   I am a beginner in MTEX. Here is a question-How can I pick out a subset of grains with GOS less than 2° for IPF or other? Here is my code:
```matlab
  [grains,ebsd.grainId] = calcGrains(ebsd('indexed'));
  ebsd(grains(grains.grainSize <= 5)) = [];
  grains(grains.grainSize < grains.boundarySize / 2) = [];
  [grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[2*degree, 10*degree]);
 grains = smooth(grains,1);
 hold on
 grod = ebsd('Titanium cubic').calcGROD(grains);
 GOS = grainMean(ebsd('Titanium cubic'), grod.angle, grains('Titanium cubic'));
 plot(grains('Titanium cubic'), GOS ./ degree)
 hold on
 condition = GOS < 2*degree;
 plot(grains(condition).boundary,'linecolor','r','linewidth',2)
 hold off
```
Does anyone know how to fix this? Thanks very much!

Best regards,
A naive phd

**Comment by ralfHielscher** [comment]
Hi Sing,

you code looks quite perfect what is the problem with it?

Ralf.

**Reply by SingPr** [reply]
Hi Ralf
     Yeah, it is work, but I wanna select some grains and mark as a subset, and plot the pole figure of the selected grains, not all grain of the EBSD. So how do I write code that selects grains of interest? and plot the PF of these grains.
thanks
Sing

**Comment by ralfHielscher** [comment, marked as answer]
Maybe something like this
```matlab
plotPDF(grains.meanOrientation, Miller(1,0,0, grains.CS))
hold on
plotPDF(grains(condition).meanOrientation, 'MarkerColor','red')
hold off
```
Ralf.

**Comment by SingPr** [comment]
Thanks, Ralf., I get it.
